### ISSUE TITLE
Do you really keep track of the type of array keys?

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#945]
If the error was on a variable and I absolutely knew that the variable was an array and not a string I could @var array $variable and the inspection error would disappear.  How do I set the typeinfo for the $ar['z'] element?

```
<?php


function x(y)
{
    /* @var string[][] $ys */
    $ys = [];
   
    $ay = $ys[0];

    $ar = [];
    $ar['z'] = $ay;
    // [] operator not supported for strings
    $ar['z']['zz'][] = [];
}
```

Ahh, I have to @var the $ay variable to fix the $ar['z'] type info.

**Comment by kalessil** [comment]
Hey, @iakoder : in this case, we are relying on the IDE type resolver, so the proper type hinting should o the trick. Looks like you found a solution already, or?

**Comment by iakoder** [comment]
It just seemed that you knew more about the type of $ar['z'] than I would have expected.  The inspection can't just be about the type of $ar because that's just an array.  The IDE had to have carried the type info of $ay into $ar['z'] to make that inspection fire on $ar['z']['zz'][] fire.  I've never seen PhpStorm pretend to know about the types of array elements which made me think that you might be extending the type info yourself.

But yea, while creating this I was able to figure out that I needed to change the type info of $ay.
